pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        """
+        Implements a server reactor for
+        :meth:`pywbem.WBEMConnection.SetQualifier`.
+
+        Create or modify a qualifier type in the local repository of this
+        class.  This method will create a new namespace for the qualifier
+        if none is defined.
+
+        Return: TODO
+
+        Exceptions:
+            CIMError CIM_ERR_INVALID_PARAMETER or
+            CIMError(CIM_ERR_ALREADY_EXISTS
+        """
+        qual = params['QualifierDeclaration']
+
+        if namespace not in self.qualifiers:
+            self.qualifiers[namespace] = NocaseDict({})
+
+        if not isinstance(qual, CIMQualifierDeclaration):
+            raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                           'The qualifier declaration parameter is not a '
+                           'valid CIMQualifierDeclaration: %s' % qual)
+
+        if qual.name in self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_ALREADY_EXISTS, '%s' % qual.name)
+        try:
+            self.qualifiers[namespace][qual.name] = qual
+        except KeyError:
+            self.qualifiers[namespace] = NocaseDict({qual.name: qual})
+
+    def _fake_deletequalifier(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.DeleteQualifier`
+
+        Deletes a single qualifier if it is in the
+        repository for this class and namespace
+
+        Exceptions;
+            CIMError CIM_ERR_INVALID_NAMESPACE, CIMError(CIM_ERR_NOT_FOUND
+        """
+        if not self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+
+        qname = params['QualifierName']
+
+        if qname in self.qualifiers[namespace]:
+            del self.qualifiers[namespace][qname]
+        else:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Qualifier decl %s not found.'
+                           % qname)
+
+    #####################################################################
+    #
+    #  Faked WBEMConnection Instance methods
+    #
+    #####################################################################
+
+    def _fake_createinstance(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.CreateInstance`
+
+        Create a CIM instance in the local repository of this class.
+
+        Exceptions:
+            CIMError CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_CLASS
+        """
+
+        new_inst = params['NewInstance']
+
+        # Requires corresponding class to build path to be returned
+        if not self._class_exists(namespace, new_inst.classname):
+            raise CIMError(CIM_ERR_INVALID_CLASS, 'Cannot create instance '
+                                                  'without corresponding class '
+                                                  '%s.' % new_inst.classname)
```

Comment:
I would prefer a message like "Cannot create instance because its creation class {classname} does not exist in namespace {namespace}".

Reply:
Done